piet/src/gradient.rs

```diff
@@ -1,22 +1,44 @@
 //! Gradient specifications.
+//!
+//! We provide both linear and radial gradients; and for each flavor
+//! we provide two representations, a 'generic' representation that uses
+//! points in the [unit square], and a 'fixed' representation that uses
+//! image-space coordinates.
+//!
+//! The generic representations ([`LinearGradient`] and [`RadialGradient`])
+//! are strictly more flexible than the fixed representations
```

Comment:
I'm not 100% sure I agree with this "strictly more flexible." I consider them sibling use cases. You'll use the unit square for UI, but the fixed ones for dealing with content in existing formats such as SVG. I think this doc is clear enough without the editorial. Probably what I'd say is "a convenience for use cases such as UI." The part after the semicolon is fine.